src/misc.c

```diff
@@ -10,11 +10,35 @@
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
 
 static int ishexdigit(const char);
 static unsigned char hexdigit2byte(char);
 static char byte2hexdigit(unsigned char);
 
+int
+local_rng(u8 *output, size_t bytes)
```

Comment:
If I remember correctly: The whole thing about internal generation if the IV is because FIPS requires to use a FIPS-approved random number generator, right? So is /dev/p|urandom a FIPS approved random number generator? I guess not....

We could probably use OpenSSL's random number generator, but then we have a dependency on OpenSSL just because of that .....

Reply:
Yes exactly. I didn't want to introduce a dependency to openssl at this time. The used local_rng implementation is also used in opencryptoki, so it should be "good enough for now", but probably "subject to change". And libzpc is still far away from any real fips compliance. Even when using the openssl RAND_bytes() function, this is not automatically fips compliant.

Reply:
Yes I know ... in a fips environment we would enforce the internal creation of the iv. Here, we just provide the opportunity to do this. However, when creating an internal iv, the application would not need to call an own random number generator, i.e. also not care about which one to use. 
And applications using this technique would behave at least a little bit in a fips manner.
But yes, they are still allowed to use the already existing API. 
This item is just a little enhancement towards fips, but not more.